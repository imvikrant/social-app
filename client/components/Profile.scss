@import '@material/layout-grid/mixins';

.profile {
  // background: #e5e5e5;
  height: calc(100vh - 64px);
  overflow-y: auto;
  padding: 0 !important;
  @include mdc-layout-grid(desktop, 0px, 1000px);

  // .row {
  //   @include mdc-layout-grid-inner($gutter: 5px, $size: 'desktop', $margin: 5px);
  // }

  

  .heading {
    padding: 5px 15px;
  }

  &-cover {
    
    height: 200px;
    background-size: cover;
    background-position: center;
    position: relative;
  }

  &-photo {
    object-fit: cover;
    border: 5px solid #fff;
    height: 150px;
    width: 150px;
    position: absolute;
    bottom: 0;
    border-radius: 50%;
    transform: translate(50%, 50%)
  }

  .posts {
    margin-top: 50px;
  }

  .post {
    width: 100%;
    margin: 20px auto;
  }

}