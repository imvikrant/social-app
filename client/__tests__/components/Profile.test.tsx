import React from 'react';
import { shallow } from 'enzyme';
import Profile from '../../components/Profile';

let wrapper: any;

beforeEach(() => {
  wrapper = shallow(<Profile match={{params: {username: 'vikrant'}}}/>)
})

test('should render Profile correctly', () => {
  expect(wrapper).toMatchSnapshot();
})