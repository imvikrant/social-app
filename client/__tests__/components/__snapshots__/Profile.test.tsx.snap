// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Profile correctly 1`] = `
<Query
  query={
    Object {
      "definitions": Array [
        Object {
          "directives": Array [],
          "kind": "OperationDefinition",
          "name": undefined,
          "operation": "query",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "alias": undefined,
                "arguments": Array [],
                "directives": Array [],
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "users",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "username",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "age",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "gender",
                      },
                      "selectionSet": undefined,
                    },
                  ],
                },
              },
            ],
          },
          "variableDefinitions": Array [],
        },
      ],
      "kind": "Document",
      "loc": Object {
        "end": 65,
        "start": 0,
      },
    }
  }
>
  <Component />
</Query>
`;
